tservice/sys/types/build.rs

```diff
@@ -2,51 +2,39 @@
 //! Builds the FFI type bindings for tservice, (trusted service) of the Intel
 //! SGX SDK
 
-use bindgen::{callbacks::ParseCallbacks, Builder};
-
-#[derive(Debug)]
-struct Callbacks;
-
-impl ParseCallbacks for Callbacks {
-    fn item_name(&self, name: &str) -> Option<String> {
-        match name {
-            "_aes_gcm_data_t" => Some("sgx_aes_gcm_data_t".to_owned()),
-            "_sealed_data_t" => Some("sgx_sealed_data_t".to_owned()),
-            name => {
-                if name.starts_with("_sgx") || name.starts_with("_tee") {
-                    Some(name[1..].to_owned())
-                } else {
-                    None
-                }
-            }
-        }
-    }
-}
+const SERVICE_TYPES: &[&str] = &[
+    "_aes_gcm_data_t",
+    "_sealed_data_t",
+    "_sgx_dh_msg1_t",
+    "_sgx_dh_msg2_t",
+    "_sgx_dh_msg3_body_t",
+    "_sgx_dh_msg3_t",
+    "_sgx_dh_session_enclave_identity_t",
+    "_sgx_dh_session_role_t",
+    "_sgx_dh_session_t",
+    "_sgx_report2_mac_struct_t",
+    "_tee_cpu_svn_t",
+    "_tee_measurement_t",
+    "_tee_report_data_t",
+    "_tee_report_type_t",
+    "tee_mac_t",
+];
```

Comment:
Should we be binding the other data types defined by `sgx_report2.h` here? It's only included by `sgx_util.h` and `internal/arch.h` from the SDK.

Reply:
There's also `align_req_t` defined in `sgx_secure_align_api.h`, which as I commented elsewhere declares certain functions from `tservice.a`.

Reply:
Added the report 2 header so if/when we get to automating it "just works"